### ISSUE TITLE
Connecting to server when client has unsaved changes always results in out of sync document

### ISSUE THREAD

**Issue by Ruin0x11** [Ruin0x11/intellij-lsp-server#40]
The server has no way to know of the client's unsaved changes when it connects, because it currently operates on the pretext that the file on disk is the ground truth when a file is opened. There should be a way of updating IDEA's ground truth inside the `textDocument/didOpen` request, since the entire document text is passed in.

However, when trying to set IDEA's document text to the passed in text, the file/memory conflict resolution window appears every time the client saves (#39). There are no issues with synchronization, and choosing `Show Differences` shows that the contents are identical. But if the user forgets to close the window the entire server will become blocked.

It would be best to override the resolver window's behavior to never appear, but this seems infeasible due to the class's accessibility paired with the need to extend it. For the sake of not having to manipulate the resolver window outside of the client on every save, it would be better for now to just avoid sending the `textDocument/didOpen` request when the client has unsaved changes. This bug only really happens when sending a command like `lsp-restart-workspace` for development purposes when the client has unsaved changes.

**Comment by Ruin0x11** [comment]
Fixed by #44 at the expense of causing #39.

**Comment by jojojames** [comment]
I did see something similar to this today with the tip of head. https://github.com/jojojames/intellij-lsp-server/commit/49f227401a030995ec622c8b643d23783adb8000

Not sure how I reproduced it but the dialog kept coming up about showing differences. It might've been when the project was still indexing. I'll have to find some time to get a sample-repro project up. Using this with proprietary code for now.
